grammars/css.cson

```diff
@@ -528,251 +1662,355 @@
-          '5':
-            'name': 'keyword.operator.pattern.css'
-          '6':
-            'name': 'string.unquoted.attribute-value.css'
-          '7':
-            'name': 'string.quoted.attribute-value.css'
-          '8':
-            'name': 'punctuation.definition.string.begin.css'
-          '9':
-            'name': 'punctuation.definition.string.end.css'
-          '10':
-            'name': 'punctuation.definition.entity.css'
         'match': '''(?x)
-          (\\[)\\s*
-          (?:(\\g<ident>|\\*)?(\\|))?     # qualified name prefix
-          ([\\w:&&[^0-9]][\\w:.-]*+)\\s*  # attribute name
-          (?:
-            ([~|^$*]?=)\\s*
-            (?:                           # attribute value
-              (\\g<ident>)
-            | ((?>([\'"])(?:[^\\\\]|\\\\.)*?(\\k<-2>)))
-            )\\s*
+          (?:^|(?<=[\\s,(};]))         # Follows whitespace, comma, semicolon, or bracket
+          (?!
+            [-\\w*]+
+            \\|
+            (?!
+                [-\\[:.*\\#a-zA-Z_]    # Make sure there's a selector to match
+              | [^\\x00-\\x7F]
+            )
+          )
+          (
+            (?: [-a-zA-Z_]    | [^\\x00-\\x7F] )   # First letter
+            (?: [-a-zA-Z0-9_] | [^\\x00-\\x7F]     # Remainder of identifier
+              | \\\\(?:[0-9a-fA-F]{1,6}|.)
+            )*
+            |
+            \\*     # Universal namespace
           )?
-          (\\])
-          (?<ident>-?+[-a-zA-Z_\\P{ASCII}][-a-zA-Z0-9_\\P{ASCII}]*+){0}
+          (\\|)     # Namespace separator
         '''
-        'name': 'meta.attribute-selector.css'
+      }
+      {
+        'include': '#tag-names'
+      }
+      {
+        'match': '\\*'
+        'name': 'entity.name.tag.wildcard.css'
       }
       {
         'captures':
           '1':
             'name': 'punctuation.definition.entity.css'
+          '2':
+            'patterns': [
+              {
+                'include': '#escapes'
+              }
+            ]
         'match': '''(?x)
-          (:)
+          (\\.)
           (
-            link|target|visited                                     # location
-          | active|focus|hover                                      # user action
-          | default|disabled|enabled|read-only|read-write           # input control state
-          | checked|indeterminate                                   # input value state
-          | in-range|invalid|optional|out-of-range|required|valid   # input value-checking
-          | empty|(first|last|only)-(child|of-type)|root            # tree-structural
-          | first|left|right                                        # @page
-          | host                                                    # shadow DOM
-          )(?![\\w-])
+            -?+                             # possessive to avoid matching '-' followed by [0-9]
+            (?![0-9])
+            (?:[-a-zA-Z_0-9]|[^\\x00-\\x7F]|\\\\(?:[0-9a-fA-F]{1,6}|.))+
```

Comment:
Again, if we are to support escapes, we should also take care of the whitespace separating the seqences and normal characters—this is not good enough. Still, we can't expect to make things perfect in one PR. Up to you if you wanna take it into subsequent PRs.